### PR TITLE
glibc: update to 2.35.

### PR DESCRIPTION
**Pull request by ericonr**
From release notes:

  Starting with glibc 2.34, the shared objects are installed under their
  ABI sonames directly, without symbolic links.

Therefore, we have to rename many of the nostrip entries.

Also from release notes:

  all functionality formerly implemented in the libraries libpthread,
  libdl, libutil, libanl has been integrated into libc.

Therefore, we can remove these libraries from the nostrip list entirely.

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/glibc/template [hunk 1]
```diff
@@ -20,27 +20,22 @@ nostrip_files="
...
-	libc-${version}.so
-	libBrokenLocale-${version}.so
-	libnss_compat-${version}.so
-	libnss_db-${version}.so
-	libnss_dns-${version}.so
-	libnss_files-${version}.so
-	libnss_hesiod-${version}.so
-	libnss_nisplus-${version}.so
-	libnss_nis-${version}.so"
+	ld-linux-x86-64.so.2
```
Comment (ericonr) [inline srcpkgs/glibc/template:23]: Suggestions on how to cover all dynamic linkers? Or just list them extensively? Can this take a glob?

**Comment by JamiKettunen** [comment]
Could this be bumped to [2.35](https://sourceware.org/pipermail/libc-alpha/2022-February/136040.html) which released about 2 weeks ago?

**Comment by ericonr** [comment]
Arch has patched glibc heavily since 2.35 release so idk how I feel about pushing it. https://github.com/archlinux/svntogit-packages/commits/packages/glibc/trunk

**Comment by wundrweapon** [comment]
Might be worth changing the PR title to reflect the version change; easier to search for that way (I almost didn't find this)

File: common/shlibs [hunk 2]
```diff
@@ -49,6 +49,35 @@ libcidn.so.1 glibc-2.32_1
...
+libnss_files.so.2 glibc-2.34_1
+libnss_compat.so.2 glibc-2.34_1
+libnss_dns.so.2 glibc-2.34_1
+libnss_hesiod.so.2 glibc-2.34_1
+libcrypt.so.1 glibc-2.34_1
+libBrokenLocale.so.1 glibc-2.34_1
+libmemusage.so glibc-2.34_1
+libSegFault.so glibc-2.34_1
+libpcprofile.so glibc-2.34_1
+libmvec.so.1 glibc-2.34_1
```
Comment (wundrweapon) [inline common/shlibs:80]: If you're updating to 2.35, do these not need to be adjusted as well? i.e. not point to 2.34

**Comment by oreo639** [comment]
glibc 3.36 is out. (the only patch Arch has for that so far is just a workaround for an Easy Anticheat issue)

**Comment by AngryPhantom** [comment]
Any plans updating it, guys? It's still 2.32 which is two years old. And what does _Arch_ have to do with glibc btw?

**Comment by JamiKettunen** [comment]
This should probably be closed; it seems `glibc` update ([to 2.36](https://github.com/void-linux/void-packages/pull/34902/commits/d55530d4ebd7a97aee4b0e9ed947c8cabb695225)) will be done as a part of #34902 (`gcc: update to 12.2.0.`) unless I'm mistaken.

**Comment by AngryPhantom** [comment]
@JamiKettunen Ok, thanks!

**Comment by oreo639** [comment]
AngryPhantom:
> And what does Arch have to do with glibc btw?

ericonr:
> Arch has patched glibc heavily since 2.35 release so idk how I feel about pushing it.

**Comment by AngryPhantom** [comment]
@oreo639 I saw this. I mean what Arch has to do with glibc which is [GNU.org](https://www.gnu.org/software/libc/) ? Does Void ship glibc patched by Arch? Why if so?

**Comment by AngryPhantom** [comment]
@oreo639 Oh, thanks for the answer (and your patience). Didn't know that glibc is that buggy, jeez :(

**Comment by AngryPhantom** [comment]
What about looking into Debian testing or something?

**Comment by AngryPhantom** [comment]
And what about backward [compatibility](https://developers.redhat.com/blog/2019/08/01/how-the-gnu-c-library-handles-backward-compatibility)?

**Comment by AngryPhantom** [comment]
> Arch has patched glibc heavily since 2.35 release so idk how I feel about pushing it. https://github.com/archlinux/svntogit-packages/commits/packages/glibc/trunk

Haha... Have you seen what [Debian](http://deb.debian.org/debian/pool/main/g/glibc/glibc_2.31-13+deb11u4.debian.tar.xz) does to it? Arch is a school kid compared to Debian.

**Comment by oreo639** [comment]
> And what about backward compatibility?

Usually it's fine. That doesn't mean issues can't occasionally slip through the cracks.

> Have you seen what Debian does to it? Arch is a school kid compared to Debian...

It was just an example.

**Pull request closed without merging**